12-make_cov.py

```diff
@@ -57,8 +58,21 @@ def run_covariance(subject, session=None):
     epochs = mne.read_epochs(fname_epo, preload=True)
 
     print('  Computing regularized covariance')
-    cv = KFold(3, random_state=config.random_state)  # make cv deterministic
-    cov = mne.compute_covariance(epochs, tmax=0, method='shrunk', cv=cv)
+    if isinstance(config.random_state, np.random.RandomState):
+        msg = (f'Using custom random number generator in cross-validation '
+               f'scheme: {config.random_state}')
+    elif config.random_state is not None:
+        msg = (f'Cross-validation random number generator will be seeded with '
+               f'the user-defined value: {config.random_state}')
+    else:
+        msg = 'Using random number generator with default settings'
+
+    print(msg)
+    del msg
+
+    cv = KFold(3, random_state=config.random_state, shuffle=True)
+    cov = mne.compute_covariance(epochs, tmax=0, method='shrunk', cv=cv,
+                                 rank='info')
```

Comment:
rank='info' is bug fix but the rest is just doc right?

Reply:
and `shuffle=True` in the `KFold` instantiation.

FYI this PR contains two commits, one for the bugfix and one for the docs & logging:
bugfix: 2069391
doc: e928ea9

Reply:
arff I the present case (with temporally correlated data) I wouldn't not shuffle. You don't want in the test set samples that are "just" like train because of autocorrelated noise. Please keep shuffle=False and make a note. This technique is also known as "block cross-validation" in stats.

Reply:
Wow I didn't know that! Thanks for the clarification, I'm glad you caught this mistake!

I've reverted those commits (including the documentation & logging updates, which are unnecessary now). I've added `shuffle=False` (which is the default in `sklearn`, but I think being explicit here is a good thing), and added a comment trying to explain _why_ we don't want to shuffle the data.

 Can you please review again?